hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
> We never expect test_packages to be empty

Yes, but that's not true about `$test_flags`. I'd prefer a solution that work with them in the similar way. Using bash is already way too tricky and black-magicky for the kind of things we're trying to do.

I would really like to use something in the lines you proposed, I tried simpler things yesterday but they all had some flaw.

```console
$ cat empty-array.sh 
#!/bin/bash
set -o nounset
test_flags=()
echo ${test_flags[@]}
$ ./empty-array.sh 
./empty-array.sh: line 4: test_flags[@]: unbound variable
```